pkg/operation/botanist/health_check.go

```diff
@@ -651,14 +674,15 @@ func (b *Botanist) checkControlPlane(
 // checkSystemComponents checks whether the system components of a Shoot are running.
 func (b *Botanist) checkSystemComponents(
 	ctx context.Context,
+	gardenerVersion string,
 	checker *HealthChecker,
 	condition gardencorev1beta1.Condition,
 	shootDeploymentLister kutil.DeploymentLister,
 	shootDaemonSetLister kutil.DaemonSetLister,
 	extensionConditions []ExtensionCondition,
 ) (*gardencorev1beta1.Condition, error) {
 
-	if exitCondition, err := checker.CheckSystemComponents(metav1.NamespaceSystem, condition, shootDeploymentLister, shootDaemonSetLister); err != nil || exitCondition != nil {
+	if exitCondition, err := checker.CheckSystemComponents(metav1.NamespaceSystem, b.Shoot.KonnectivityTunnelEnabled, condition, shootDeploymentLister, shootDaemonSetLister); err != nil || exitCondition != nil {
```

Comment:
Again, this will fail for 1.18 clusters that weren't reconciled yet. As a reference, you could check in the etcd-druid PR #1762 what was done.